### PR TITLE
[FIX] crm: fill company fields in workflow without lead step

### PR DESCRIPTION
**Pull request by yelizariev**
TLDR: "Convert to Opportunity" button fills Company field in partner, while with
disabled Lead step the partner is created without Company value

STEPS:

* install crm
* disable Lead step in Settings
* create opportunity with fields: Opportunity name, Customer Name (Company),
Contact name, Email, but without Customer field (this is equal to creating a
lead via website_crm)
* send a message to get popup "Please complete customer's informations"

BEFORE: company value is not set

AFTER: company value contains newly created company record

WHY:

* ``with_context(active_model=None, active_id=None)``: to avoid infinite loop
with creating parents
* ``vals["email"] = False``: otherwise *Customer* value is set to Company
record, rather than to Contact record
* this way we have another problem, but I don't see how to easily fix it in
stable branch. The problem:
  * if creating *Contact* is canceled, then we have created company, but not a
Contact.
  * trying to send message again will create another company record

---

opw-2365942

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by nim-odoo** [comment]
Indeed, it looks quite hacky with a change of behavior. I'll check this in details tomorrow morning.

**Review by nim-odoo — COMMENTED** [review]
It is in line with what is done in:
https://github.com/odoo/odoo/blob/272602193f5647f7f2270ed6ec68777625a139dd/addons/crm/models/crm_lead.py#L840-L841

I don't see any other way to do it, but conceptually having a `default_get` calling a `create` is not good. In your example, we can do:
- Click on "Send Message"
- Cancel the contact creation

=> a partner is created for the company, but no partner is created for the contact. That doesn't sound very good.

So I'd go for a no-fix in this case.

**Comment by yelizariev** [comment]
Another idea is adding ``default_name`` to company creation context, so at least user doesn't need to copy-paste it

**Comment by nim-odoo** [comment]
> Another idea is adding `default_name` to company creation context, so at least user doesn't need to copy-paste it

Not sure of what you mean. Could you update the PR?

**Comment by yelizariev** [comment]
How it works: 
https://drive.google.com/file/d/1OStjrP5lZcAmzkW6CHEE6yAxhmiNyp91/view?usp=sharing

**Comment by yelizariev** [comment]
commit description is updated too

**Comment by nim-odoo** [comment]
Wow ok, never seen that before :smile: I'd still go for a "no-fix", I'd rather avoid this kind of hacky things for non-blocking issues.

**Comment by yelizariev** [comment]
:crying_cat_face:
